### ISSUE TITLE
'Make' using install tree

### ISSUE THREAD

**Issue by luojinzhang** [oci-labs/node-opendds#44]
Hi,
So when I try to `make` the project with setenv.sh, it built and tested successfully. However, when I try to set it to installedOpenDDS/share/dds (from dds-devel.sh), it does not build, the error was:
```make[1]: Entering directory '/home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/test/idl'
NodeJSTest will not be built due to the following missing library: OpenDDS_Dcps

GNUmakefile: /home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/test/idl/GNUmakefile.NodeJSTest MAKEFLAGS=w

make[1]: Leaving directory '/home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/test/idl'
make[1]: Entering directory '/home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/tools/v8stubs'

GNUmakefile: /home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/tools/v8stubs/GNUmakefile.v8stubs MAKEFLAGS=w

make[1]: Leaving directory '/home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/tools/v8stubs'
make[1]: Entering directory '/home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/test'
test_publisher will not be built due to the following missing library: OpenDDS_Tcp OpenDDS_InfoRepoDiscovery OpenDDS_Dcps

GNUmakefile: /home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/test/GNUmakefile.test_publisher MAKEFLAGS=w

make[1]: Leaving directory '/home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/test'
make[1]: Entering directory '/home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/test'
test_subscriber will not be built due to the following missing library: OpenDDS_Tcp OpenDDS_InfoRepoDiscovery OpenDDS_Dcps

GNUmakefile: /home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/test/GNUmakefile.test_subscriber MAKEFLAGS=w

make[1]: Leaving directory '/home/laccc/Projects/demoCycloneOpenDDS-JS/node-opendds/test'```

**Comment by mitza-oci** [comment]
I didn't see the same error when I tried this.  If you want to dig in to the specifics you can run `make LIBCHECK=1` and post the compile or link command lines it's trying to use.

The projects in the `test/` and `tools/` directories of this repository are not designed to work with the installed OpenDDS layout.  They're for internal testing, so supporting this doesn't seem necessary.  Of course if someone wanted to add code to make it work, that would be a welcome contribution.

I confirmed that the npm module itself does build using the installed OpenDDS layout.

**Comment by simpsont-oci** [comment]
I was able to reproduce this on my machine and I was able to get a bit further by adding the install prefix `lib` directory to the test `.mpc` files using the `libpath` command, but as Adam points out these tests weren't designed to be run from an installed OpenDDS. Specifically, they make use of utility headers from the OpenDDS repository's `tests` directory which aren't copied over as part of the install process, as well as using the PerlACE and PerlDDS perl modules as part of run_test, which are also not installed as part of the install process. At some point in the future it might be nice to support using an install for these tests, but it would take a bit of effort to migrate away from these utilities.
